crates/engine/tree/src/persistence.rs

```diff
@@ -38,55 +39,24 @@ impl<Writer> Persistence<Writer>
 where
     Writer: Unpin,
 {
-    /// Internal method to poll the persistence task. This returns [`None`] if the channel for
-    /// incoming [`PersistenceAction`]s is closed.
-    #[tracing::instrument(level = "debug", name = "Persistence::poll", skip(self, cx))]
-    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<PersistenceOutput>> {
-        let this = self.get_mut();
-
-        if let Some(handle) = this.active_writer_thread {
-            // if pending we just keep waiting until we can do something
-            if let Err(_err) = ready!(handle.poll_unpin(cx)) {
-                todo!("handle errors");
-            }
-
-            this.active_writer_thread.take();
-        }
-
-        let action = match ready!(this.incoming.poll_recv(cx)) {
-            None => return Poll::Ready(None),
-            Some(action) => action,
-        };
-
-        match action {
-            PersistenceAction::RemoveBlocksAbove(new_tip_num) => {
-                let (sender, receiver) = oneshot::channel();
-
-                // spawn blocking so we can poll the thread later
-                let handle = spawn_blocking(move || {
+    /// This is the main loop, that will listen to persistence events and perform the requested
+    /// database actions
+    async fn run(&mut self) {
+        // TODO: sync or async receiver?
```

Comment:
we expect that this will only do IO so this should be spawned on a regular endless thread, like we do with the mdbx db

see also: https://ryhl.io/blog/async-what-is-blocking/

Reply:
makes sense, switched to sync loop